core/client/fs/src/main/java/alluxio/client/file/MetadataCachingBaseFileSystem.java

```diff
@@ -73,12 +79,78 @@ public MetadataCachingBaseFileSystem(FileSystemContext context) {
         TimeUnit.SECONDS, new SynchronousQueue<>());
   }
 
+  @Override
+  public void createDirectory(AlluxioURI path, CreateDirectoryPOptions options)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, AlluxioException {
+    super.createDirectory(path, options);
+    mMetadataCache.invalidate(path.getParent());
+    mMetadataCache.invalidate(path);
+  }
+
+  @Override
+  public FileOutStream createFile(AlluxioURI path)
+      throws IOException,
+      AlluxioException {
+    FileOutStream outStream = super.createFile(path);
+    mMetadataCache.invalidate(path.getParent());
+    mMetadataCache.invalidate(path);
+    return outStream;
+  }
+
+  @Override
+  public FileOutStream createFile(AlluxioURI path, CreateFilePOptions options)
+      throws IOException,
+      AlluxioException {
+    FileOutStream outStream = super.createFile(path, options);
+    mMetadataCache.invalidate(path.getParent());
+    mMetadataCache.invalidate(path);
+    return outStream;
+  }
+
+  @Override
+  public void delete(AlluxioURI path)
```

Comment:
override this method, which is the default impl of `delete`
```suggestion
  rename(AlluxioURI src, AlluxioURI dst, RenamePOptions options)
```